go/ql/lib/semmle/go/frameworks/GoMicro.qll

```diff
@@ -0,0 +1,159 @@
+/**
+ * Provides models of the [Go Micro library](https://github.com/go-micro/go-micro).
+ */
+
+import go
+private import semmle.go.security.RequestForgeryCustomizations
+
+/**
+ * Module for Go Micro framework.
+ */
+module GoMicro {
+  /**
+   * A GoMicro server type.
+   */
+  class GoMicroServerType extends Type {
+    GoMicroServerType() { this.hasQualifiedName("go-micro.dev/v4/server", "Server") }
+  }
+
+  /**
+   * A GoMicro client type.
+   */
+  class GoMicroClientType extends Type {
+    GoMicroClientType() { this.hasQualifiedName("go-micro.dev/v4/client", "Client") }
+  }
+
+  /**
+   * A file that is generated by the protobuf compiler.
+   */
+  class ProtocGeneratedFile extends File {
+    ProtocGeneratedFile() { this.getBaseName().regexpMatch(".*\\.pb(\\.micro)?\\.go$") }
+  }
+
+  /**
+   * A type that is generated by the protobuf compiler.
+   */
+  class ProtocMessageType extends Type {
+    ProtocMessageType() {
+      exists(TypeEntity te |
+        te.getType() = this and
+        te.getDeclaration().getLocation().getFile() instanceof ProtocGeneratedFile and
+        exists(MethodDecl md |
+          md.getName() = "ProtoMessage" and
+          this = md.getReceiverType().(PointerType).getBaseType()
+        )
+      )
+    }
+
+    override string getName() { result = this.getUnderlyingType().getName() }
```

Comment:
I think this is wrong. It is giving no result for the two instances of this class. Can I ask what you were trying to do here? The instances of this class are `NamedType`s (because methods are only defined on named types) so they have a perfectly good `getName()`. Their underlying types are the struct types they are aliases for, which don't have a name. So I think this predicate should just be deleted.

Reply:
Removed code